app/themes/dashboard/ui/app/navigation/navigation.js

```diff
@@ -66,18 +69,29 @@
         $state.go(link.state);
       };
 
+      service.showSidebar = true;
+
+      service.toggleSidebar = function () {
+        service.showSidebar = !service.showSidebar;
+      };
+
       return service;
     }
 
-    TopNavCtrl.$injector = ['$scope', 'userService'];
-    function TopNavCtrl($scope, userService) {
-      var ctrl = this;
+  TopNavCtrl.$injector = ['$scope', 'userService', '$document', 'loginService'];
 
-      ctrl.currentUser = userService.getUser();
+  function TopNavCtrl($scope, userService, $document, loginService) {
+    var ctrl = this;
 
-      $scope.$watch(userService.currentUser, function(newValue) {
-        ctrl.currentUser = newValue;
-      });
-    }
+    ctrl.currentUser = userService.getUser();
+
+    $scope.$watch(userService.currentUser, function(newValue) {
+      ctrl.currentUser = newValue;
+    });
+
+    ctrl.logout = function() {
+      loginService.logout();
+    };
+  }
 
 })();
```

Comment:
I was adding userService to the root controller already in the other PR. How about exposing the loginService as well, and just getting rid of TopNavCtrl altogether. You can expose user and loginService in the generic root controller to make them generally available..